src/context/AppProvider/hooks/useAccountsFetchQuery.tsx

```diff
@@ -0,0 +1,209 @@
+import { usePrevious } from '@chakra-ui/react'
+import type { AccountId, ChainId } from '@shapeshiftoss/caip'
+import { fromAccountId } from '@shapeshiftoss/caip'
+import { isLedger } from '@shapeshiftoss/hdwallet-ledger'
+import { MetaMaskShapeShiftMultiChainHDWallet } from '@shapeshiftoss/hdwallet-shapeshift-multichain'
+import type { AccountMetadataById } from '@shapeshiftoss/types'
+import { skipToken, useQuery } from '@tanstack/react-query'
+import { useCallback, useEffect, useMemo } from 'react'
+import { usePlugins } from 'context/PluginProvider/PluginProvider'
+import { useIsSnapInstalled } from 'hooks/useIsSnapInstalled/useIsSnapInstalled'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { walletSupportsChain } from 'hooks/useWalletSupportsChain/useWalletSupportsChain'
+import { deriveAccountIdsAndMetadata } from 'lib/account/account'
+import { isUtxoChainId } from 'lib/utils/utxo'
+import { portfolio, portfolioApi } from 'state/slices/portfolioSlice/portfolioSlice'
+import { selectEnabledWalletAccountIds } from 'state/slices/selectors'
+import { txHistoryApi } from 'state/slices/txHistorySlice/txHistorySlice'
+import { useAppDispatch, useAppSelector } from 'state/store'
+
+export const useAccountsFetchQuery = () => {
+  const dispatch = useAppDispatch()
+  const enabledWalletAccountIds = useAppSelector(selectEnabledWalletAccountIds)
+  const { supportedChains } = usePlugins()
+  const { isSnapInstalled } = useIsSnapInstalled()
+  const previousIsSnapInstalled = usePrevious(isSnapInstalled)
+  const { deviceId, wallet } = useWallet().state
+
+  // TODO(gomes): this was wrong and still is. selectEnabledWalletAccountIds will populate accountIds as they start getting enabled
+  // This is currently fixed by not including hasManagedAccounts in the queryKey deps, as well as explicitly disabling the queryFn if there are managed accounts and
+  // having a separate effect as a side-effect, but ideally, we should have an *actual* mechanism to detect managed accounts, since regular accounts discovery also goes through the same process
```

Comment:
@woodenfurniture FYI - would like to discuss this one for a follow-up.
This makes things much, much better and safer as-is but side-effect will still run (though as a clear side-effect now), even without custom account management. Should be a no-op though, but conceptually was risky and still is. 

The reason is, as written in the comment, there is no clear distinction between manually enabled/disabled AccountIds and enabled AccountIds as part of the regular autodiscovery flow.

Thinking one very naive fix here would be as soon as you manually manage accounts (enable a new/disable an account), we set a `isAccountManagementEnabled` toggle on in state, WDYT?

Reply:
Yes considering the potential for future work surrounding all of this I'm strongly in favor of a simple patch here, like the one you describe above.

Reply:
Will do in a follow-up, ty for the confirmation ser!

Reply:
Ended up with way simpler of a solution in the end @woodenfurniture! https://github.com/shapeshift/web/pull/7932